blaze/utils.py

```diff
@@ -266,3 +272,20 @@ def parameter_space(*args):
     return tuple(product(*(
         arg if isinstance(arg, tuple) else (arg,) for arg in args
     )))
+
+
+def memoize_mutable(fn):
```

Comment:
Why not make the function returrn a function to defer construction. pickle is not designed for speed / data efficiency, it is just supposed to work. imagine something like:

```python
@dec
def f():
    a = 1
    b = 2
    return lambda: {'a': a, 'b': b}
```

`dec` could call the nullary function every time the value is requested to rebuild a new value.

I am not sure I love the name memoize mutable since the function returns a new object and I normally think of memoize as `f(x) is f(x)`